### PR TITLE
Add Service Principal for image sync

### PR DESCRIPTION
**Pull request by janboll**
Add a service principal intended for image sync processes. It requires accesss to the service key vault to read pull secrets. It additionally requires access to the ACR (Push).

### What this PR does

Jira: 
Link to demo recording:

### CODE REVIEW

File: .github/workflows/bicep-what-if.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
               --name "${GITHUB_RUN_ID}" \
               --resource-group "${SC_RESOURCEGROUP}" \
               --template-file templates/dev-acr.bicep \
-              --parameters configurations/dev-acr.bicepparam
+              --parameters configurations/mvp-dev-acr.bicepparam
```
Comment (janboll) [inline .github/workflows/bicep-what-if.yml:50]: This is  a fix for earlier refactoring #394

File: dev-infrastructure/modules/acr-push-permission.bicep [hunk 2]
```diff
@@ -0,0 +1,19 @@
+param principalName string
```
Comment (geoberle) [inline dev-infrastructure/modules/acr-push-permission.bicep:1]: can we refactor the acr-push-permission and acr-pull-permission modules into one and and parameterize on the permission? e.g. have a look at the keyvault-secret-access module

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 3]
```diff
@@ -404,12 +404,13 @@ resource acrRg 'Microsoft.Resources/resourceGroups@2023-07-01' existing = [
...
 ]
 
-module acrPullRole 'acr-pull-permission.bicep' = [
+module acrPullRole 'acr-permissions.bicep' = [
   for (_, i) in acrResourceGroups: {
     name: guid(acrRg[i].id, aksCluster.id, acrPullRoleDefinitionId)
     scope: acrRg[i]
     params: {
       principalId: aksCluster.properties.identityProfile.kubeletidentity.objectId
+      grantPushAccess: true
```
Comment (bennerv) [inline dev-infrastructure/modules/aks-cluster-base.bicep:413]: Should we rename this module since it now can grant push access too?

Comment (weinong) [reply dev-infrastructure/modules/aks-cluster-base.bicep:413]: why do we allow it to push to ACR?

**Comment by weinong** [comment]
I'm fine using kubelet's identity to pull and push images _for now_ until we enable workload identity on https://github.com/Azure/msi-acrpull and bootstrap it in these clusters. However, before we onboard any real customers I need a complete isolation from the svc cluster that runs the service. 

cc: @bennerv

File: dev-infrastructure/modules/aks-cluster-base.bicep (hunk 3 above)
Comment (weinong) [reply dev-infrastructure/modules/aks-cluster-base.bicep:413]: i think i understand what you want to accomplish. please use https://github.com/Azure/ARO-HCP/pull/398#issuecomment-2251842547 to continue the discussion

**Comment by bennerv** [comment]
> I'm fine using kubelet's identity to pull and push images _for now_ until we enable workload identity on https://github.com/Azure/msi-acrpull and bootstrap it in these clusters. This will allow us to decouple image pull from node's IMDS/kubelet MSI so that we can manage it by namespace. However, before we onboard any real customers I need a complete isolation from the svc cluster that runs the service.
> 
> cc: @bennerv
 
👍 

I raised this last week.  We have a card (https://issues.redhat.com/browse/ARO-9404) to change it later down the road before we hit Prod.

File: dev-infrastructure/modules/aks-cluster-base.bicep (hunk 3 above)
Comment (janboll) [reply dev-infrastructure/modules/aks-cluster-base.bicep:413]: This push is here by mistake. I added it for testing, purposes. Let me remove it please.

**Review by geoberle — APPROVED** [review]

**Pull request merged**